projects/dotcms/src/lib/api/DotApiForm.ts

```diff
@@ -27,32 +38,78 @@ export class DotApiForm {
      * @memberof DotApiForm
      */
     async render(container: HTMLElement) {
-        this.fields =
-            this.fields || (await this.dotApiContentType.getFields(this.formConfig.identifier));
+        this.contentType =
+            this.contentType || (await this.dotApiContentType.get(this.formConfig.identifier));
+        this.fields = this.contentType.fields;
 
-        const fieldScript = this.createFieldTags(this.fields);
-        const formScript = this.createForm(fieldScript);
         const importScript = document.createElement('script');
-        const formTag = document.createElement('div');
-
         importScript.type = 'module';
         importScript.text = `
-            import { defineCustomElements } from "https://unpkg.com/dotcms-field-elements@0.0.2/dist/loader";
+            import { defineCustomElements } from 'http://localhost:8080/fieldElements/loader/index.js';
+            //import { defineCustomElements } from 'https://unpkg.com/dotcms-field-elements@0.0.2/dist/loader';
             defineCustomElements(window);`;
-        formTag.innerHTML = formScript;
 
+        const fieldScript = this.createFieldTags(this.fields);
+        const formTag = this.createForm(fieldScript);
         container.append(importScript, formTag);
     }
 
-    private createForm(fieldTags: string): string {
-        return `<form>${fieldTags}</form>`;
+    private shouldSetFormLabel(label: string, labelConfig: {submit?: string, reset?: string}): boolean {
+        return !!(labelConfig && labelConfig[label]);
+    }
+
+    private createForm(fieldScript: string): HTMLElement {
+        const formTag = document.createElement('dot-form');
+
+        formTag.setAttribute(
+            'submit-label',
+            this.shouldSetFormLabel('submit', this.formConfig.labels)
+                ? this.formConfig.labels.submit
+                : 'Submit'
+        );
+        formTag.setAttribute(
+            'reset-label',
+            this.shouldSetFormLabel('reset', this.formConfig.labels)
+                ? this.formConfig.labels.reset
+                : 'Reset'
+        );
+        formTag.innerHTML = fieldScript;
+
+        formTag.addEventListener('formSubmit', (e: CustomEvent) => {
```

Comment:
This is one of the things I think we should dig into it... we're creating a form type custom element but the submit event is a custom event, can we make it a submit event? this comments is based on the stencil doc:

> There is NOT such a thing as stencil events, instead, Stencil encourages the use of DOM events.